File: src/console-model/Environment.py
from Input import Settings
from time import clock
from Output import Output


class Environment:
	# This class handles the configuration files
	# of the model, as well as output for results
	
	# Publics
	RunSettings = None
	TownSettings = None
	Output = None
	OutputSummary = None
	
	# Privates
	_startTime = 0
	
	# Constructor
	def __init__ (self):
		self.Output = Output("results.txt")
		self.OutputSummary = Output("resultsSummary.txt")
		self.RunSettings = Settings("run.ini").LoadSection("General")
		self.TownSettings = Settings("town.ini").LoadSection("General")
		self._startTime = clock()
		return

	# Execution elapsed time indicator
	def getElapsedTime(self):
		return clock() - self._startTime
	ElapsedTime = property(getElapsedTime)

File: src/interactive-simulator/CasaresModel/Environment.cs
using System;
using System.Collections;
using System.Collections.Specialized;
using System.Drawing;

namespace CasaresModel
{
	public class Environment
	{
		// This class handles the configuration files
		// of the model, as well as output for results
		
		// Publics
		public Hashtable RunSettings;
		public Hashtable TownSettings;
		public Output Output;
		public Output OutputSummary;
		
		// Privates
		public DateTime _startTime;
		
		// Constructor
		public Environment ()
		{
			Output = new Output("results.txt");
			OutputSummary = new Output("resultsSummary.txt");
			RunSettings = Settings.LoadSection("run.ini", "General");
			TownSettings = Settings.LoadSection("town.ini", "General");
			_startTime = DateTime.Now;
		}
		
		// Execution elapsed time indicator
		public TimeSpan getElapsedTime()
		{
			return DateTime.Now - _startTime;
		}
		public TimeSpan	ElapsedTime 
		{
			get
			{
				return getElapsedTime();
			}
		}
	}
}
